src/VisualStudio/Core/Next/SymbolSearch/VisualStudioSymbolSearchService.ProgressService.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.SymbolSearch;
+using Microsoft.VisualStudio.TaskStatusCenter;
+using Roslyn.Utilities;
+using VSShell = Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.LanguageServices.SymbolSearch
+{
+    [ExportWorkspaceService(typeof(ISymbolSearchProgressService), ServiceLayer.Host), Shared]
+    internal class VisualStudioSymbolSearchProgressService : ISymbolSearchProgressService
+    {
+        private readonly object _gate;
+        private readonly IVsTaskStatusCenterService _taskCenterServiceOpt;
+
+        private TaskCompletionSource<bool> _taskCompletionSource;
+
+        [ImportingConstructor]
+        public VisualStudioSymbolSearchProgressService(VSShell.SVsServiceProvider serviceProvider)
+        {
+            _taskCenterServiceOpt = new ProgressService((IVsTaskStatusCenterService)serviceProvider.GetService(typeof(SVsTaskStatusCenterService)));
```

Comment:
Thanks.  I opened a VSO bug against you about this.  Feel free to direct it accordingly.  I'll make this lazy in the meantime.